users/ignoring.rst

```diff
@@ -88,6 +88,32 @@ The ``.stignore`` file contains a list of file or path patterns. The
    are *included* (that is, *not* ignored). This can be used to override
    more general patterns that follow.
 
+   .. note::
+
+      Negated patterns that can match items below the folder root will cause
+      Syncthing to traverse otherwise ignored directories. If the
```

Comment:
This first sentence requires serious mental capacity to parse, even for me as an advanced user.  Better to split it into smaller statements:

"Syncthing will sometimes look at the files even inside ignored directories.  Some file that matches any negated pattern (thus, should be *included*) could very well reside somewhere below them.  This happens only for directories matching an ignore pattern *after the first negated pattern*."

Then the example, with speaking directory names that directly show what happens:

```
/1-not_looked_at
/2-also_completely_disregarded
!3-filename_to_sync_anyway
!4-foo
/5-could_have_foo_below_so_scan_it
*
```

Not that this is the ideal solution, but hopefully easier to grasp.  So I hope you don't mind me just sketching the concept without following through.